op-e2e/actions/garbage_channel_out.go

```diff
@@ -0,0 +1,246 @@
+package actions
+
+import (
+	"bytes"
+	"compress/gzip"
+	"compress/zlib"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type GarbageKind int64
+
+const (
+	STRIP_VERSION GarbageKind = iota
+	RANDOM
+	TRUNCATE_END
+	DIRTY_APPEND
+	INVALID_COMPRESSION
+	MALFORM_RLP
+)
+
+var GarbageKinds = []GarbageKind{
+	STRIP_VERSION,
+	RANDOM,
+	TRUNCATE_END,
+	DIRTY_APPEND,
+	INVALID_COMPRESSION,
+	MALFORM_RLP,
+}
+
+// GarbageChannelCfg is the configuration for a `GarbageChannelOut`
+type GarbageChannelCfg struct {
+	useInvalidCompression bool
+	malformRLP            bool
+}
+
+// WriterApi is the interface shared between `zlib.Writer` and `gzip.Writer`
+type WriterApi interface {
```

Comment:
Going to defer to @trianglesphere and @protolambda re: naming of interfaces, i know people are opinionated on it and i don't believe we use the `Api` suffix for interfaces currently. More opinionated about it being standard

Reply:
yea, I'd avoid putting `API` on the end. We sometimes use `Iface` or `Interface`.
Here I'd use something like `Writer` or `WriteCloser` or `WriteFlushResetCloser` (this is the golang tradition of stuffing everything together: https://pkg.go.dev/io#WriteCloser).  We rely on the package name to separate the different types.

Reply:
Changed:
* `WriterApi` -> `Writer`
* `ChannelOutApi` -> `ChannelOutIface`